src/sentry/incidents/models.py

```diff
@@ -34,9 +36,41 @@
 from sentry.models.team import Team
 from sentry.services.hybrid_cloud.user.service import user_service
 from sentry.snuba.models import QuerySubscription
+from sentry.snuba.subscriptions import delete_snuba_subscription
 from sentry.utils import metrics
 from sentry.utils.retries import TimedRetryPolicy
 
+alert_subscription_callback_registry: dict[
+    AlertRuleMonitorType, Callable[[QuerySubscription], bool]
+] = {}
+
+
+def register_alert_subscription_callback(
+    monitor_type: AlertRuleMonitorType,
+) -> Callable[[Callable], Callable]:
+    def decorator(func: Callable) -> Callable:
+        alert_subscription_callback_registry[monitor_type] = func
+        return func
+
+    return decorator
+
+
+def invoke_alert_subscription_callback(
+    monitor_type: AlertRuleMonitorType | int, subscription: QuerySubscription
```

Comment:
Does this need `| int`?

thinking this should either be the monitor type (and we pull out the `.value` when we need the int) or we just keep it a straight `int`

Reply:
this implementation allows you to pass `AlertRuleMonitorType.ACTIVATED` or `AlertRuleMonitorType.ACTIVATED.value` -- i was just building in some optionality with how to work with the enum, it feels a bit wonky to be passing things around as `.value` vs the enum to me 🤷